source/AI.cpp

```diff
@@ -3286,20 +3286,22 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 			if(shift)
 				ship.SetTargetShip(shared_ptr<Ship>());
 			
-			double closest = numeric_limits<double>::infinity();
+			double best = numeric_limits<double>::infinity();
 			bool foundEnemy = false;
 			bool foundAnything = false;
+			bool distancePriority = Preferences::Has("Board target");
 			for(const shared_ptr<Ship> &other : ships)
 				if(CanBoard(ship, *other))
 				{
 					if(shift && !other->IsYours())
 						continue;
 					
 					bool isEnemy = other->GetGovernment()->IsEnemy(ship.GetGovernment());
-					double d = other->Position().Distance(ship.Position());
-					if((isEnemy && !foundEnemy) || (d < closest && isEnemy == foundEnemy))
+					double distance = other->Position().Distance(ship.Position());
+					double b = distancePriority ? distance : -other->Cost() / (distance / 200);
```

Comment:
Does this give a division by zero if the distance is zero?

Reply:
that's fiiiine, just board that ship damnit we're on top of it (never had the case of it being 0)
more seriously yeah that may be an issue, I'll fix it, well spotted